src/mono/wasm/runtime-test.js

```diff
@@ -5,9 +5,34 @@
 
 //glue code to deal with the differences between chrome, ch, d8, jsc and sm.
 var is_browser = typeof window != "undefined";
+var consoleWebSocket;
+var print;
 
 if (is_browser) {
 	// We expect to be run by tests/runtime/run.js which passes in the arguments using http parameters
+	window.real_print = console.log;
+	print = function(_msg) { window.real_print(_msg); };
+	console.log = print;
+	console.debug = print;
+	console.error = print;
+	console.trace = print;
+	console.warn = print;
+	console.info = print;
+
+	const consoleUrl = `${window.location.origin}/console`.replace('http://', 'ws://');
+
+	consoleWebSocket = new WebSocket(consoleUrl);
+	consoleWebSocket.onopen = function(event) {
+		consoleWebSocket.send("browser: Console websocket connected.");
```

Comment:
This might be a better place to force everything through print?  I sometimes use runtime-tests.js outside of xharness

Reply:
Yeah, that makes sense. And I was thinking maybe the WebSocket thing should be optional? I found it useful to run the tests without the harness, directly in a browser.

Reply:
I think it is worth a follow-up once we have everything working in CI.